server/src/main/scala/slickdb/DropIns.scala

```diff
@@ -0,0 +1,106 @@
+                     isPublished: Boolean,
+                     meetingLink: Option[String],
+                     lastUpdatedAt: Timestamp) {
+  def toDropIn: DropIn = DropIn(id, title, startTime.getTime, endTime.getTime, isPublished, meetingLink, lastUpdatedAt.getTime)
+
+  def getDate: String = getUKStringDate(startTime, dateFormatter)
+
+  def getStartTime: String = getUKStringDate(startTime, timeFormatter)
+
+  def getEndTime: String = getUKStringDate(endTime, timeFormatter)
+
+  val dateFormatter: DateTimeFormatter = DateTimeFormatter.ofPattern("dd/MM/yyyy")
+
+  val timeFormatter: DateTimeFormatter = DateTimeFormatter.ofPattern("HH:mm")
+
+  def getUKStringDate(timestamp: Timestamp, formatter: DateTimeFormatter): String = zonedUKDateTime(timestamp).format(formatter)
+
+  val zonedUKDateTime: Timestamp => ZonedDateTime = timestamp => timestamp.toInstant.atZone(ZoneId.of("Europe/London"))
+
+}
+
+class DropIns(tag: Tag) extends Table[DropInRow](tag, "drop_in") {
+  def id: Rep[Option[Int]] = column[Option[Int]]("id", O.PrimaryKey, O.AutoInc)
+
+  def title: Rep[String] = column[String]("title")
+
+  def startTime: Rep[Timestamp] = column[Timestamp]("start_time")
+
+  def endTime: Rep[Timestamp] = column[Timestamp]("end_time")
+
+  def isPublished: Rep[Boolean] = column[Boolean]("is_published")
+
+  def meetingLink: Rep[Option[String]] = column[Option[String]]("meeting_link")
+
+  def lastUpdatedAt: Rep[Timestamp] = column[Timestamp]("last_updated_at")
+
+  def * : ProvenShape[DropInRow] = (id, title, startTime, endTime, isPublished, meetingLink, lastUpdatedAt).mapTo[DropInRow]
+}
+
+trait DropInTableLike {
+  def updatePublishDropIn(dropInId: String, publish: Boolean): Future[Int]
+
+  def updateDropIn(dropInRow: DropInRow): Future[Int]
+
+  def deleteDropIn(dropInId: String): Future[Int]
+
+  def getFuturePublishedDropIns()(implicit ec: ExecutionContext): Future[Seq[DropIn]]
+
+  def getDropIns(ids: Seq[String])(implicit ec: ExecutionContext): Future[Seq[DropInRow]]
+}
+
+case class DropInTable(tables: Tables) extends DropInTableLike {
+  val dropInTable = TableQuery[DropIns]
+
+  private def getCurrentTime = new Timestamp(new DateTime().getMillis)
+
+  def updatePublishDropIn(dropInId: String, publish: Boolean): Future[Int] = {
+    val query = dropInTable.filter(_.id === dropInId.trim.toInt).map(f => (f.isPublished, f.lastUpdatedAt))
+      .update(publish, getCurrentTime)
+    tables.run(query)
+  }
+
+  def updateDropIn(dropInRow: DropInRow): Future[Int] = dropInRow.id match {
+    case Some(id) =>
+      val query = dropInTable.filter(_.id === id).map(f => (f.title, f.startTime, f.endTime, f.lastUpdatedAt))
+        .update(dropInRow.title, dropInRow.startTime, dropInRow.endTime, getCurrentTime)
+      tables.run(query)
+    case None => Future.successful(0)
+  }
+
+  def deleteDropIn(dropInId: String): Future[Int] = {
+    val query = dropInTable.filter(_.id === dropInId.trim.toInt).delete
+    tables.run(query)
+  }
```

Comment:
It just occurred to me that this app doesn't update or delete dropins, so we don't need these functions